pybop/_problem.py

```diff
@@ -269,23 +275,16 @@ def evaluate(self, x):
             Parameter values to evaluate the model at.
         """
 
-        y = np.asarray(self._model.simulate(inputs=x, t_eval=self._time_data))
-
-        return y
-
-    def evaluateS1(self, x):
-        """
-        Evaluate the model with the given parameters and return the signal and its derivatives.
-
-        Parameters
-        ----------
-        x : np.ndarray
-            Parameter values to evaluate the model at.
-        """
-
-        y, dy = self._model.simulateS1(
+        sol = self._model.predict(
             inputs=x,
-            t_eval=self._time_data,
+            experiment=self.experiment,
+            init_soc=self.init_soc,
         )
 
-        return (np.asarray(y), np.asarray(dy))
+        if sol == [np.inf]:
+            return sol
+
+        else:
+            predictions = [sol[signal].data for signal in self.signal + ["Time [s]"]]
```

Comment:
I thought this logic was included in BaseModel now, but perhaps I'm incorrect. Did we add this in the previous PR from this branch?

Reply:
I think the conflict is that we also use `model.predict` to return a PyBaMM solution so, to keep this functionality, I think the unpacking of the solution has to be done at this level?